include/eos/fitting/ceres_nonlinear.hpp

```diff
@@ -331,33 +355,35 @@ struct ImageCost
             // Note: We could store the BiCubicInterpolator as member variable.
             // The default template arguments for Grid2D are <T, kDataDim=1, kRowMajor=true,
             // kInterleaved=true> and (except for the dimension), they're the right ones for us.
-            ceres::Grid2D<uchar, 3> grid(image.ptr(0), 0, image.rows, 0, image.cols);
-            ceres::BiCubicInterpolator<ceres::Grid2D<uchar, 3>> interpolator(grid);
-            T observed_colour[3];
+            auto grid = ceres::Grid2D<uchar, 3>(image.ptr(0), 0, image.rows, 0, image.cols);
```

Comment:
Here I also think `auto` makes the code harder to read but I guess that's just a personal preference and style.

Reply:
Your repo - your rules :-) It is just about code style